### PR TITLE
Benchmark multi batch indexing

### PR DESCRIPTION
**Pull request by Kerollmops**
Hey @irevoire, could you please add the new benchmarks into influx?

### CODE REVIEW

**Review by irevoire — APPROVED** [review]
Looks good to me!
Lets just wait for [this action](https://github.com/meilisearch/milli/actions/runs/1881844365) to finish before merging in case we missed something.

Also I'll be able to add the benchmarks in influxdb only after we merged this PR on main and waited for the benchmarks to execute, so probably tomorrow :+1:

File: benchmarks/benches/indexing.rs [hunk 1]
```diff
@@ -83,6 +83,77 @@ fn indexing_songs_default(c: &mut Criterion) {
...
+                let faceted_fields =
+                    ["released-timestamp", "duration-float", "genre", "country", "artist"]
+                        .iter()
+                        .map(|s| s.to_string())
+                        .collect();
+                builder.set_filterable_fields(faceted_fields);
+                builder.execute(|_| ()).unwrap();
+
+                // We index only one half of the dataset in the setup part
+                // as we don't care about the time it take.
```
Comment (curquiza) [inline benchmarks/benches/indexing.rs:119]:
```suggestion
                // as we don't care about the time it takes.
```

File: benchmarks/benches/indexing.rs [hunk 2]
```diff
@@ -223,6 +294,76 @@ fn indexing_wiki(c: &mut Criterion) {
...
+                builder.set_displayed_fields(displayed_fields);
+
+                let searchable_fields = ["title", "body"].iter().map(|s| s.to_string()).collect();
+                builder.set_searchable_fields(searchable_fields);
+
+                // there is NO faceted fields at all
+                builder.execute(|_| ()).unwrap();
+
+                // We index only one half of the dataset in the setup part
+                // as we don't care about the time it take.
```
Comment (curquiza) [inline benchmarks/benches/indexing.rs:321]:
```suggestion
                // as we don't care about the time it takes.
```

**Comment by irevoire** [comment]
Look like the action failed because the benchmark took too much time...
> The job running on runner benchmark has exceeded the maximum execution time of 360 minutes.

I thought our limitation was 48h @curquiza ? :thinking: 
I guess we'll have to split the re-indexation to another file :pensive:

**Review by irevoire — APPROVED** [review]
After #455 was merged we were able to run the benchmarks, see [this action](https://github.com/meilisearch/milli/runs/5305800044?check_suite_focus=true) that ran for 9 hours I'm totally up for merging this PR!
Thanks @Kerollmops :guide_dog:

**Comment by curquiza** [comment]
@irevoire @Kerollmops here are the expected limits: https://docs.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners#usage-limits

**Pull request merged**